iocage/lib/Datasets.py

```diff
@@ -0,0 +1,146 @@
+import helpers
+
+import libzfs
+
+
+class Datasets:
+
+    ZFS_POOL_ACTIVE_PROPERTY = "org.freebsd.ioc:active"
+
+    def __init__(self, root=None, pool=None, zfs=None, logger=None):
+        helpers.init_logger(self, logger)
+        helpers.init_zfs(self, zfs)
+
+        self._datasets = {}
+
+        if isinstance(root, libzfs.ZFSDataset):
+            self.root = root
+            return
+
+        if isinstance(pool, libzfs.ZFSPool):
+            self.root = self._get_or_create_dataset(
+                "iocage",
+                root_name=pool.name,
+                pool=pool
+            )
+            return
+
+        active_pool = self.active_pool
+
+        if active_pool is None:
+            msg = ("iocage is not activated yet - "
+                   "please run `iocage activate` first and select a pool")
+            self.logger.error(msg)
+            raise Exception(msg)
+        else:
+            self.root = self.zfs.get_dataset(f"{active_pool.name}/iocage")
+                
+
+    @property
+    def active_pool(self):
+        prop = self.ZFS_POOL_ACTIVE_PROPERTY
+        for pool in self.zfs.pools:
+            try:
+                if pool.root_dataset.properties[prop].value == "yes":
+                    return pool
+            except:
+                pass
```

Comment:
When the property `prop` does not exist on `pool.root_dataset.properties` an exception is thrown, that we want to ignore.

```
>>> a = { "foo": 123 }
>>> a["bar"]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
KeyError: 'bar'
```